### PR TITLE
bpo-37322: Fix ResourceWarning and exception handling in test (GH-25553)

### PR DESCRIPTION
**Pull request by tiran**
Handle all OSErrors in a single block. OSError also takes care of
SSLError and socket's connection errors.

Partly reverts commit fb7e7505ed1337bf40fa7b8b68317d1e86675a86. The
threaded connection handler must not raise an unhandled exception.

https://bugs.python.org/issue37322

### CODE REVIEW

**Comment by tiran** [comment]
Buildbots failed to trigger, let's try again.

File: Lib/test/test_ssl.py [hunk 1]
```diff
@@ -2470,31 +2470,18 @@ def run(self):
...
-                        )
-                    self.close()
-                    self.running = False
-                except ssl.SSLError as err:
-                    # On Windows sometimes test_pha_required_nocert receives the
-                    # PEER_DID_NOT_RETURN_A_CERTIFICATE exception
-                    # before the 'tlsv13 alert certificate required' exception.
-                    # If the server is stopped when PEER_DID_NOT_RETURN_A_CERTIFICATE
-                    # is received test_pha_required_nocert fails with ConnectionResetError
-                    # because the underlying socket is closed
```
Comment (matthewhughes934) [inline Lib/test/test_ssl.py:2501]: Is there a risk of this failing on Windows, or would this be covered by the CI suite? I'm not too familiar with the code (nor windows) so I kept away from this when I was looking around this issue.

Comment (tiran) [reply Lib/test/test_ssl.py:2501]: Python has pre-commit CI and post-commit buildbots for all major platforms including multiple flavors of Windows.

It doesn't make any sense the raise an exception here. The exception just breaks the handler thread and does not even end up in the test case.

Comment (matthewhughes934) [reply Lib/test/test_ssl.py:2501]: Thanks! That clears things up

**Pull request merged**